src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,113 @@
+                if ((instruction[1].contains(" /by "))
+                        && (instruction[1].split(" /by ")[1]
+                                .matches("^\\d{4}-(0[1-9]|1[012])\\-(0[1-9]|[12][0-9]|3[01]) "
+                                        + "([01]?[0-9]|2[0-3]):[0-5][0-9]"))) {
+                    return new DeadlineCommand(instruction[1].split(" /by ")[0],
+                            instruction[1].split(" /by ")[1]);
+
+                } else {
+                    throw new DukeException("Please input in the following format "
+                            + "'deadline <description> /by <yyyy-MM-dd HH:mm>' with a valid date & time");
+                }
+            }
+
+        } else if (instruction[0].equals("event")) {
+
+            if (instruction.length == 1) {
+                throw new DukeException("☹ OOPS!!! The description of a event cannot be empty.");
+            } else {
+                //check if contains event " /at yyyy-MM-dd HH:mm"
+                if ((instruction[1].contains(" /at "))
+                        && (instruction[1].split(" /at ")[1]
+                                .matches("^\\d{4}-(0[1-9]|1[012])-(0[1-9]|[12][0-9]|3[01]) "
+                                        + "([01]?[0-9]|2[0-3]):[0-5][0-9]"))) {
+                    return new EventCommand(instruction[1].split(" /at ")[0],
+                            instruction[1].split(" /at ")[1]);
+                } else {
+                    throw new DukeException("Please input in the following format "
+                            + "'event <description> /at <yyyy-MM-dd HH:mm>' with a valid date & time");
+                }
+            }
+        } else if (instruction[0].equals("done")) {
+            //done with no other arguments
+            if (instruction.length == 1) {
+                throw new DukeException("Please specify item number!");
+            } else if (instruction.length == 2) { //done with exactly 2 inputs
+                return new DoneCommand(instruction[1]);
+
+            }
+        } else if (instruction[0].equals("delete")) {
+            //done with no other arguments
+            if (instruction.length == 1) {
+                throw new DukeException("Please specify item number!");
+            } else if (instruction.length == 2) { //done with exactly 2 inputs
+                return new DeleteCommand(instruction[1]);
+            }
+        } else if (instruction[0].equals("find")) {
+            //find with no other arguments
+            if (instruction.length == 1) {
+                throw new DukeException("Please specify a description to search!");
+            } else {
+                return new FindCommand(instruction[1]);
+            }
+        } else {
+            throw new DukeException("☹ OOPS!!! I'm sorry, but I don't know what that means :-(");
+        }
+
+        throw new DukeException("☹ OOPS!!! I'm sorry, but I don't know what that means :-(");
+    }
```

Comment:
Perhaps try to shove some code into separate methods? This method seems way too long (although many others do the same).